internal/retry/retry_test.go

```diff
@@ -0,0 +1,159 @@
+// Copyright 2018 The Go Cloud Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package retry
+
+import (
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	gax "github.com/googleapis/gax-go"
+)
+
+// Used to indicate a duration returned from the isRetryable function.
+// gax.Backoff.Pause never returns negative durations.
+const retryDur = time.Duration(-1)
+
+// Errors to distinguish retryable and non-retryable cases.
+var (
+	errRetry   = errors.New("retry")
+	errNoRetry = errors.New("no retry")
+)
+
+func retryable(err error) (bool, time.Duration) {
+	if err == errRetry {
+		return true, retryDur
+	}
+	return false, 0
```

Comment:
What about a test for retryable returning a non-zero `Duration`?

Reply:
Those tests exist below. Look for `wantRetryableDur`.